python/oneflow/test/graph/test_graph_image_gpu_decoder.py

```diff
@@ -0,0 +1,100 @@
+"""
+Copyright 2020 The OneFlow Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import os
+import unittest
+import numpy as np
+
+import oneflow as flow
+import oneflow.unittest
+
+
+class OFRecordDataLoader(flow.nn.Module):
+    def __init__(self):
+        super().__init__()
+        batch_size = 4
+        image_size = 224
+        self.train_record_reader = flow.nn.OfrecordReader(
+            "/dataset/imagenette/ofrecord",
+            batch_size=batch_size,
+            data_part_num=1,
+            part_name_suffix_length=5,
+            random_shuffle=True,
+            shuffle_after_epoch=True,
+            # placement=flow.placement("cpu", {0: [0]}),
+            # sbp=[flow.sbp.broadcast]
+        )
+
+        self.record_label_decoder = flow.nn.OfrecordRawDecoder(
+            "class/label", shape=(), dtype=flow.int32
+        )
+        self.bytes_decoder = flow.nn.OFRecordBytesDecoder("encoded")
+        self.image_gpu_decoder = flow.nn.OFRecordImageGpuDecoderRandomCropResize(
+            target_width=image_size, target_height=image_size, num_workers=3
+        )
+
+        color_space = "RGB"
+        output_layout = "NHWC"
+
+        self.flip = flow.nn.CoinFlip(
+            batch_size=batch_size,
+            # placement=flow.placement("cpu", {0: [0]}),
+            # sbp=[flow.sbp.broadcast]
+        )
+
+        rgb_mean = [123.68, 116.779, 103.939]
+        rgb_std = [58.393, 57.12, 57.375]
+        self.crop_mirror_norm = flow.nn.CropMirrorNormalize(
+            color_space=color_space,
+            output_layout=output_layout,
+            mean=rgb_mean,
+            std=rgb_std,
+            output_dtype=flow.float,
+        )
+
+    def forward(self) -> (flow.Tensor, flow.Tensor):
+        train_record = self.train_record_reader()
+        label = self.record_label_decoder(train_record)
+        encoded = self.bytes_decoder(train_record)
+        image = self.image_gpu_decoder(encoded)
```

Comment:
Check failed: device_tag == GetDeviceTagOfTensor(input_tensor)  Lazy nn.Graph name : GraphReader_0 encountered ERROR where multi-input tensor has different device type in module/op_name: my_reader    .crop_mirror_norm-crop_mirror_normalize_from_uint8_5. Please use tensor.to() or tensor.to_consistent() to make all input with same device. 

测试失败了，image没device信息

(INPUT:_my_reader.crop_mirror_norm-input_0:tensor(..., is_lazy='True', size=(4, 224, 224, 3), dtype=oneflow.uint8))                                                                                     
(INPUT:_my_reader.crop_mirror_norm-input_1:tensor(..., device='cuda:0', is_lazy='True', size=(4,), dtype=oneflow.int8))

Reply:
我在 21 机器上是可以的。 CI 也能过。

```console
chengcheng@oneflow-21:~/debug/graph $ python3 test_graph_image_gpu_decoder.py 
Nothing happened because environment has been initialized
image tensor meta:  tensor(..., device='cuda:0', is_lazy='True', size=(4, 224, 224, 3),
       dtype=oneflow.uint8)
```

你的测试环境是 16？

Reply:
是的